lms/djangoapps/instructor_task/tasks_helper/grades.py

```diff
@@ -307,8 +314,40 @@ def __init__(self, _xmodule_instance_args, _entry_id, course_id, _task_input, ac
         self.course_id = course_id
         self.report_for_verified_only = generate_grade_report_for_verified_only()
         self.task_progress = TaskProgress(self.action_name, total=None, start_time=time())
-        self.course = get_course_by_id(self.course_id)
         self.file_name = 'problem_grade_report'
+        self.log_additional_info = six.text_type(self.course_id) == 'course-v1:MITx+CTL.SC0x+1T2019'
+
+    @lazy
+    def course(self):
+        return get_course_by_id(self.course_id)
```

Comment:
Is it courseoverview? Why are you making it lazy?

Reply:
No, its a call to the modulestore for the course. 

I am making this `lazy` to reuse the course structure from the cache rather than loading it everytime for a new batch of users.

Reply:
`context.course` is used inside `_rows_for_users` which executes for for all batches. If we cache it, we don't have to load the course from `ms` for each batch of users.

Reply:
Ok. Do you know why BS isn't the priority lookup? Just wondering why both BS and modulestore courses are going hand-in-hand in the overall report generation.

Reply:
The course is needed because we need to access the grading policy while generating the report. 

e.g. here. https://github.com/edx/edx-platform/blob/9cf2f9f298e5e8be3b3abcaadaf0b7a96d0de0df/lms/djangoapps/grades/context.py#L86